### PR TITLE
[v6r20] fix stub and rms

### PR DESCRIPTION
**Pull request by chaen**
Work done with @sfayer 

BEGINRELEASENOTES

*Core
CHANGE: Fixes the infamous "invalid action proposal" by speeding up the handshake and not looking up the user/group in the baseStub

RMS:
CHANGE: FowardDISET uses the owner/group of Request to execute the stub
CHANGE: owner/group of the Requests are evaluated/authorized on the server side
CHANGE: LimitedDelegation or FullDelegation are required to set requests on behalf of others -> pilot user and hosts should must them (which should already be the case)

ENDRELEASENOTES

### CODE REVIEW

File: RequestManagementSystem/Service/ReqManagerHandler.py [hunk 1]
```diff
@@ -34,183 +34,201 @@ class ReqManagerHandler( RequestHandler ):
...
-    request = Request( requestDict )
+    requestDict = json.loads(requestJSON)
+    requestName = requestDict.get("RequestID", requestDict.get('RequestName', "***UNKNOWN***"))
+    request = Request(requestDict)
+
+    # Check whether the credentials in the Requests are correct and allowed to be set
+    isAuthorized = RequestValidator.setAndCheckRequestOwner(request, self.getRemoteCredentials())
+
+    if not isAuthorized:
+      return S_ERROR("Credentials in the requests are not allowed")
```
Comment (atsareg) [inline RequestManagementSystem/Service/ReqManagerHandler.py]: DErrno.ENOAUTH

File: RequestManagementSystem/Service/ReqProxyHandler.py [hunk 2]
```diff
@@ -64,159 +68,173 @@ class ReqProxyHandler( RequestHandler ):
...
-    forwardable = self.__forwardable( requestDict )
-    if not forwardable["OK"]:
-      gLogger.warn( "putRequest: %s" % forwardable["Message"] )
+    # We only need the object to check the authorization
+    request = Request(requestDict)
+    # Check whether the credentials in the Requests are correct and allowed to be set
+    isAuthorized = RequestValidator.setAndCheckRequestOwner(request, self.getRemoteCredentials())
+
+    if not isAuthorized:
+      return S_ERROR("Credentials in the requests are not allowed")
```
Comment (atsareg) [inline RequestManagementSystem/Service/ReqProxyHandler.py]: Error code again

File: RequestManagementSystem/Service/ReqManagerHandler.py (hunk 1 above)
Comment (chaen) [reply RequestManagementSystem/Service/ReqManagerHandler.py]: good idea

**Comment by atsareg** [comment]
Why CI checks are not done for this PR ? Or this is only me who can not see them ?

**Comment by andresailer** [comment]
Sometimes triggers get lost, if you close and open the PR maybe the checks get triggered.

**Pull request merged**